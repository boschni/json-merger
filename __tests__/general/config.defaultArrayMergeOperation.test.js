const jsonMerger = require("../../dist");
const { testConfig } = require("../../__helpers__");

describe("when config.defaultArrayMergeOperation is", function () {
  test("not set combine should be used", function () {
    const defaultArrayMergeOperation = undefined;
    const object1 = {
      a: [1, 2, 3],
    };

    const object2 = {
      a: [3, 3],
    };

    const result = jsonMerger.mergeObjects(
      [object1, object2],
      testConfig({ defaultArrayMergeOperation })
    );

    expect(result).toMatchSnapshot();
  });

  test("set to replace, replace should be used", function () {
    const defaultArrayMergeOperation = "replace";
    const object1 = {
      a: [1, 2, 3],
    };

    const object2 = {
      a: [3, 3],
    };

    const result = jsonMerger.mergeObjects(
      [object1, object2],
      testConfig({ defaultArrayMergeOperation })
    );

    expect(result).toMatchSnapshot();
  });

  test("set to concat, concat should be used", function () {
    const defaultArrayMergeOperation = "concat";
    const object1 = {
      a: [1, 2, 3],
    };

    const object2 = {
      a: [3, 3],
    };

    const result = jsonMerger.mergeObjects(
      [object1, object2],
      testConfig({ defaultArrayMergeOperation })
    );

    expect(result).toMatchSnapshot();
  });

  test("set to a unsupported value, combine should be used", function () {
    const defaultArrayMergeOperation = "unsupported value";
    const object1 = {
      a: [1, 2, 3],
    };

    const object2 = {
      a: [3, 3],
    };

    const result = jsonMerger.mergeObjects(
      [object1, object2],
      testConfig({ defaultArrayMergeOperation })
    );

    expect(result).toMatchSnapshot();
  });

  test("set to a unsupported value, combine should be used", function () {
    const defaultArrayMergeOperation = "anotherUnsupportedValue=D";
    const object1 = {
      a: [1, 2, 3],
    };

    const object2 = {
      a: [3, 3],
    };

    const result = jsonMerger.mergeObjects(
      [object1, object2],
      testConfig({ defaultArrayMergeOperation })
    );

    expect(result).toMatchSnapshot();
  });
});
