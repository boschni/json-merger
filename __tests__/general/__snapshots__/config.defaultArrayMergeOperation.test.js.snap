// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when config.defaultArrayMergeOperation is not set combine should be used 1`] = `
"{
	"a": [
		3,
		3,
		3
	]
}"
`;

exports[`when config.defaultArrayMergeOperation is set to a unsupported value, combine should be used 1`] = `
"{
	"a": [
		3,
		3,
		3
	]
}"
`;

exports[`when config.defaultArrayMergeOperation is set to a unsupported value, combine should be used 2`] = `
"{
	"a": [
		3,
		3,
		3
	]
}"
`;

exports[`when config.defaultArrayMergeOperation is set to concat, concat should be used 1`] = `
"{
	"a": [
		1,
		2,
		3,
		3,
		3
	]
}"
`;

exports[`when config.defaultArrayMergeOperation is set to replace, replace should be used 1`] = `
"{
	"a": [
		3,
		3
	]
}"
`;
